File: Pythonleetcode/Leetcode135.py
class Solution(object):
    def candy(self, ratings):
        n=len(ratings)
        if n==0:
            return 0
        candy=[0]*n
        for i in range(n):
            if i==0:
                candy[i]=1
            elif ratings[i]<=ratings[i-1]:
                candy[i]=1
            else:candy[i]=candy[i-1]+1
        for i in range(n-2,-1,-1):
            if ratings[i]<=ratings[i+1]:
                continue
            elif candy[i]>candy[i+1]:
                continue
            else:
                candy[i]=candy[i+1]+1
                # print(candy)
        return sum(candy)









print(Solution().candy([1,0,2]))
print(Solution().candy([1,2,2]))
print(Solution().candy([1,3,2,2,1]))
print(Solution().candy([1,2,87,87,87,2,1]))

File: CSharpPractice/LeetcodePractice/Leetcode135.cs
public class Solution135 {
    public int Candy(int[] ratings) 
    {
        if(ratings.Length<2){return 1;}
        int[] ans = Enumerable.Repeat(1,ratings.Length).ToArray();
        for(int i=1;i<ratings.Length;i++)
        {
            if(ratings[i]>ratings[i-1]){ans[i] = ans[i-1]+1;}
        }
        for(int i=ratings.Length-2;i>=0;i--)
        {
            if(ratings[i]>ratings[i+1]){ans[i] = Math.Max(ans[i],ans[i+1]+1);}
        }
        return ans.Sum();
    }
}